platforms/allwinner-d1/d1-core/src/ccu.rs

```diff
@@ -248,12 +260,24 @@ macro_rules! impl_bgr {
     ($($MODULE:ident : ($reg:ident, $gating:ident, $reset:ident),)+) => {
         $(
             impl BusGatingResetRegister for $MODULE {
-                fn gating(ccu: &mut CCU, pass: bool) {
-                    ccu.$reg.modify(|_, w| w.$gating().bit(pass));
+                fn gating(ccu: &mut CCU, gating: BusGating) {
+                    ccu.$reg.modify(|_, w| {
+                        if gating == BusGating::Mask {
+                            w.$gating().clear_bit()
+                        } else {
+                            w.$gating().set_bit()
+                        }
```

Comment:
nit, take it or leave it: this could still be a one-liner if we wrote
```suggestion
                        w.$gating.bit(gating == BusGating::Pass);
```